### ISSUE TITLE
Support for Linux based OSs

### ISSUE THREAD

**Issue by omerferhatt** [mgth/LittleBigMouse#333]
As a developer who has been using multi-display for a long time, I recently discovered this repo. Even though it was a very basic approach, it blew my mind. I think such features should be offered by operating system developers.

From what I understand there is only support for Windows 11 and 10. Will you support other operating systems soon? If you don't have such plans, do you have any recommendations for using this application (or similar) on Linux-based (especially Debian-based) operating systems? (maybe another repo etc.)

Excellent work, thank you.

**Comment by mgth** [comment]
I switched to avalonia, and got system related functions separated from main app, in the hope that it will be done somedays, but for now I don't have time nor installed system to do it.

**Comment by flyrobot27** [comment]
This feature is natively support by Ubuntu Gnome as far as I'm aware

**Comment by omerferhatt** [comment]
Sorry, but I couldn't find it :/

**Comment by mgth** [comment]
As far as I can remember, you can achieve it by using a virtual higher resolution on smaller dpi monitors to normalise dpi, but it can lead to some blurriness effect. I don't know if there is now a more native solution.

**Comment by Xpos587** [comment]
I will look forward to it.

**Comment by Majoraslayer** [comment]
> This feature is natively support by Ubuntu Gnome as far as I'm aware

Where is it?  I've been struggling with figuring out how to fix mouse transitions between my monitors on Linux for a couple years.  My different sized monitors need different scaling, so when I scale them correctly Linux always changes the perceived screen sizes in the layout in an illogical way.  I might be able to work around it if there's some way to account for it in Ubuntu aside from the regular Display settings.

**Comment by TheInfamousAlk** [comment]
> > This feature is natively support by Ubuntu Gnome as far as I'm aware
> 
> Where is it? I've been struggling with figuring out how to fix mouse transitions between my monitors on Linux for a couple years. My different sized monitors need different scaling, so when I scale them correctly Linux always changes the perceived screen sizes in the layout in an illogical way. I might be able to work around it if there's some way to account for it in Ubuntu aside from the regular Display settings.

It's called "fractional scaling" but it's not really a replacement for littlebigmouse. It lets you "resize" monitors and scale them proportionally, which kind of does a similar function, but if you don't actually want that monitor zoomed, or you want fine control over the positioning, it's useless.

**Comment by Majoraslayer** [comment]
> > > This feature is natively support by Ubuntu Gnome as far as I'm aware
> > 
> > 
> > Where is it? I've been struggling with figuring out how to fix mouse transitions between my monitors on Linux for a couple years. My different sized monitors need different scaling, so when I scale them correctly Linux always changes the perceived screen sizes in the layout in an illogical way. I might be able to work around it if there's some way to account for it in Ubuntu aside from the regular Display settings.
> 
> It's called "fractional scaling" but it's not really a replacement for littlebigmouse. It lets you "resize" monitors and scale them proportionally, which kind of does a similar function, but if you don't actually want that monitor zoomed, or you want fine control over the positioning, it's useless.
> 
> I'm still searching for a good replacement for LBM on linux, I haven't found one that has all of the features yet.

Yeah, like you said, fractional scaling doesn't fix monitor alignment.  I have a 65" 4K TV and two 27" 1080p monitors positioned below it.  To make things look how I want, I have to scale the TV to 200% and the monitors to 100% with fractional scaling.  In Windows, this doesn't change the monitor sizes in the layout, so everything lines up pretty well.  However, Linux inexplicably treats all three monitors as the same physical size with these settings, so it's impossible to get mouse transitions aligned correctly.

Because of this, I think LittleBigMouse is even more urgently needed on Linux than Windows imo.  I did find ONE solution for it, an app called Cursr.  The only problem is that the functionality to fix this problem is locked behind a paywall, though after the struggle I've had with this problem it was worth the $14 fee.  Also, there's only one dev maintaining it, and they're very helpful if you have any issues with the software.  However, the whole Linux ecosystem would benefit from having a FOSS option to fix this problem, and LittleBigMouse has a good chance to step up and fill that void if the dev ever takes an interest in it.

**Comment by TheInfamousAlk** [comment]
I also am using Cursr now, but oh man the resizing is not intuitive. You have to entire sizes and pixel offset manually via raw numbers. No dragging, no sliding, nothing. Plus the paywall. Nothing near what LBM offered as far as user experience goes.

**Comment by Majoraslayer** [comment]
> I also am using Cursr now, but oh man the resizing is not intuitive. You have to entire sizes and pixel offset manually via raw numbers. No dragging, no sliding, nothing. Plus the paywall. Nothing near what LBM offered as far as user experience goes.

I've also had an issue where the Cursr license key flags as invalid randomly due to the way it tries to tie to a hardware ID in Linux.  I recommend joining the Discord for Cursr if you're going to use it just in case that value changes and you run into this problem as well.  The whole app is maintained by one person, and they're pretty good at working with you if you run into any issues like that. 
 
I don't mean to derail this issue thread, but wanted to add advice for this solution as it is currently the closest thing we have to a solution for LBM's functionality on Linux for comparison.

**Comment by TheInfamousAlk** [comment]
Any plans for this any time soon? Still struggling with cursr.

**Comment by mgth** [comment]
Update for everyone following this: **the Linux port is real and working** 🎉 — experimental and build-from-source for now. The mouse engine was rewritten in Rust with a native evdev/uinput backend (the same "total router" model as on Windows), developed and tested on KDE Plasma 6 / Wayland. The README now has a [Linux section](https://github.com/mgth/LittleBigMouse#linux-experimental) with build instructions (a one-liner on Arch/CachyOS). The first external builder got it running this week (#511). Feedback is very welcome — packaging and broader desktop coverage are the next steps; keeping this open to track them.

**Comment by omerferhatt** [comment]
I'd definitely like to see it in action. Thanks!

**Comment by TheInfamousAlk** [comment]
I've been waiting years for this, I'm so happy I can finally use it on my daily-driver linux PC. 

Per your plans for packaging - I'd love to see it on the AUR and maybe something like flatpak at some point too for maximum coverage. The AUR should be easier than dealing with flatpak permissions, but flatpak might reach more people not on arch-based distros.

**Comment by mgth** [comment]
Now available through AUR. Flatpak may follow.

**Comment by Rundervinkje** [comment]
I just made the full switch to Mint and I'm missing this piece of software for my QoL. Is it possible to manually install this on Linux Mint 22.3 - Cinnamon 64-bit? I would be happy to test it out.
